pkg/repository/repository.go

```diff
@@ -57,6 +59,21 @@ func NewRepository(client client.Client, repoCache RepoCache) Repository {
 	}
 }
 
+func (r *repository) GetDelivery(name string) (*v1alpha1.ClusterDelivery, error) {
+	delivery := &v1alpha1.ClusterDelivery{}
+
+	key := client.ObjectKey{
+		Name: name,
+	}
+
+	err := r.cl.Get(context.TODO(), key, delivery)
+	if err != nil && api_errors.IsNotFound(err) {
```

Comment:
it looks like some of the methods do this check, thus, performing it within repository, while others leave for higher levels to do so.

e.g., here at the reconciler level we check 

https://github.com/vmware-tanzu/cartographer/blob/eabeb980942d346c6e35c00b2fa229f084d83050/pkg/controller/delivery/reconciler.go#L53-L61

vs in workload's reconciler

https://github.com/vmware-tanzu/cartographer/blob/77ee9657b2af7dbb2d68353419f8f246da58ad5b/pkg/controller/workload/reconciler.go#L59-L66

is there a particular reason we should go with one over the other? I think it'd be good if we could keep them very a-like and adopt a single pattern 😁